presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java

```diff
@@ -38,24 +41,65 @@ public HdfsContext(ConnectorIdentity identity)
         this.schemaName = Optional.empty();
         this.tableName = Optional.empty();
         this.clientInfo = Optional.empty();
+        this.isNewTable = Optional.empty();
+        this.tablePath = Optional.empty();
     }
 
     public HdfsContext(ConnectorSession session)
     {
-        this(session, Optional.empty(), Optional.empty());
+        this(session, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
     }
 
     public HdfsContext(ConnectorSession session, String schemaName)
     {
-        this(session, Optional.of(requireNonNull(schemaName, "schemaName is null")), Optional.empty());
+        this(session, Optional.of(requireNonNull(schemaName, "schemaName is null")), Optional.empty(), Optional.empty(), Optional.empty());
     }
 
     public HdfsContext(ConnectorSession session, String schemaName, String tableName)
     {
-        this(session, Optional.of(requireNonNull(schemaName, "schemaName is null")), Optional.of(requireNonNull(tableName, "tableName is null")));
+        this(
```

Comment:
Please add the `boolean isNewTable` and the `Path tablePath` (no optional) parameters to this constructor to keep the numbers of constructors to the minimum. Currently it is very easy to create an incomplete object, that makes the entire mechanism very fragile. Ideally it would be nice to remove the other constructors in this class, but currently they are used by Raptor and FileMetastore, so this lays beyond the scope of this PR.

Reply:
- There is only one call site left for this constructor in `RaptorPageSinkProvider`, which will be deprecated. 
- I marked this constructor as deprecated.